src/gamma.jl

```diff
@@ -719,18 +719,61 @@ Euler integral of the first kind ``\\operatorname{B}(x,y) = \\Gamma(x)\\Gamma(y)
 """
 function beta end
 
-function beta(x::Real, w::Real)
-    yx, sx = logabsgamma(x)
-    yw, sw = logabsgamma(w)
-    yxw, sxw = logabsgamma(x+w)
-    return exp(yx + yw - yxw) * (sx*sw*sxw)
+function beta(a::Real, b::Real)
+    #Special case for negative integer argument
+    if a <= 0.0
+        if a == floor(a)
+            if isinteger(a)
```

Comment:
Can combine some of the `if` statements
```suggestion
            if isinteger(a) && isinteger(b) && 1-a-b > 0
```